### DISCUSSION TITLE
To get involved in this project, which knowledge should I learn first?

### DISCUSSION THREAD

**Discussion by WuMianzhi** [maplibre/maplibre-gl-js#4527] [Q&A] [answered]
In the early stages, I tried to get involved with this project and attempted to solve the issue. However, it seems too difficult for me, and I'm not sure how to proceed. Could you please advise which area of knowledge I should focus on first to contribute effectively?

**Comment by HarelM** [comment, marked as answer]
There are readme scattered around the repo, I would advise to read them. 
I have marked some issues with "good first issue".
There are developers guides as well. 
There's also a need to understand webgl, typescript and general web development and capabilities. 
Let me know if those help, and if not, what you think would help.

**Reply by WuMianzhi** [reply]
Thank you for your reply. I think I need knowledge about WebGL. Basically, I have no idea about WebGL and matrix concepts.

**Reply by sbachinin** [reply]
Webgl is good to have but there are lots of non-webgl contributions that you can make
